New APIG throttling policy resource supported

Lance52259 opened the pull request:
<!-- Thanks for sending a pull request! -->

**What this PR does / why we need it**:
The throttling policy controls the number of times APIs can be called by a user, an app, or an IP address during a specific period to protect backend services.
- For different services and users, custom can control the request frequency at which an API can be called by a user, an app, and an IP address. This ensures that backend services can run stably.
- The throttling can be accurate to the second, minute, hour, or day.
- Excluded apps and tenants can be configured to limit the number of API calls from specific apps and tenants, respectively.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
reference: #1249 

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->

```release-note
1. New throttling policy resource support:
  a. support base type and share type
  b. support four kinds of call limit
2. Related acc unit test for each parameters.
3. Related document support.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccApigThrottlingPolicyV2_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccApigThrottlingPolicyV2_basic -timeout 360m -parallel 4
=== RUN   TestAccApigThrottlingPolicyV2_basic
=== PAUSE TestAccApigThrottlingPolicyV2_basic
=== CONT  TestAccApigThrottlingPolicyV2_basic
--- PASS: TestAccApigThrottlingPolicyV2_basic (466.09s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig  466.161s
```
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccApigThrottlingPolicyV2_spec'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccApigThrottlingPolicyV2_spec -timeout 360m -parallel 4
=== RUN   TestAccApigThrottlingPolicyV2_spec
=== PAUSE TestAccApigThrottlingPolicyV2_spec
=== CONT  TestAccApigThrottlingPolicyV2_spec
--- PASS: TestAccApigThrottlingPolicyV2_spec (485.65s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      485.731s
```

Lance52259 committed:
feat: new apig throttling policy resource supported

- support base type and share type
- support four kinds of call limit
- support special throttling policies management

Lance52259 commented:
Pending sdk supports throttling policy.

chengxiangdong commented on docs/resources/apig_throttling_policy.md:
```diff
@@ -0,0 +1,137 @@
+
+* `id` - ID of the API throttling policy.
+* `user_throttles/throttling_object_name` - The object name which the special user throttling policy belongs.
+* `user_throttles/id` - ID of the special user throttling policy.
```
Suggest to write in the following format.  `app_throttles` also the same.

* `user_throttles` - Specifies an array of one or more special throttling policies for APP limit.
    Each item of data is as follows:
    -  `throttling_object_name` - The object name which the special user throttling policy belongs.
    -  `id` - ID of the special user throttling policy.

Lance52259 replied:
These texts are more readable.

chengxiangdong commented on huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy.go:
```diff
@@ -0,0 +1,458 @@
+		return fmtp.Errorf("Error saving special user throttles to state: %s", err)
+	}
+	apps := make([]map[string]interface{}, len(specThrottles))
+	for i, throttle := range specThrottles {
```
It is better to construct `users` and `apps` in one loop.

Lance52259 replied:
thx, that's easy to understand.

Lance52259 committed:
update the throttling policy provider and document

- use more readable code
- refactor the policies structure of attribute parameters

ShiChangkuo commented on docs/resources/apig_throttling_policy.md:
```diff
@@ -0,0 +1,142 @@
+  This parameter applies with each of the API call limits: `max_api_requests`, `max_app_requests`, `max_ip_requests`
+  and `max_user_requests`.
+
+* `max_api_requests` - (Required, Int) Specifies the maximum number of times an API can be accessed within a specified
```
please put  `max_api_requests`, `max_app_requests`, `max_ip_requests` and `max_user_requests` together.

ShiChangkuo commented on huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy.go:
```diff
@@ -0,0 +1,459 @@
+	return nil
+}
+
+func resourceApigThrottlingPolicyV2Create(d *schema.ResourceData, meta interface{}) error {
```
Please add some blank lines to make it easier to read.

Lance52259 committed:
update the provider and document

- adjust the locations of all api limit call in document
- add some blank lines and annotations
